### PR TITLE
feat: add CMIP7 data requirements support for Example provider

### PR DESCRIPTION
**Pull request by lewisjared**
## Description

Add CMIP7 data requirements support, enabling the Example provider to fetch CMIP6 data from ESGF and translate it to CMIP7 format.

### Key Changes

- **CMIP7Request class** (`climate_ref_core/esgf/cmip7.py`): Fetches CMIP6 data from ESGF and converts it to CMIP7 format using the CMIP7 CV converter
- **Example provider updates**: Support both CMIP6 and CMIP7 source types with flexible data requirements
- **Test cases CLI**: Add support for CMIP7 datasets in `ref test-cases` command
- **Error handling improvements**: Better messages for missing providers, graceful handling of permission errors and OR logic edge cases
- **Caching**: Use platformdirs cache for converted CMIP7 files

## Checklist

Please confirm that this pull request has done the following:

- [x] Tests added
- [x] Documentation added (where applicable)
- [x] Changelog item added to `changelog/`

### CODE REVIEW

File: packages/climate-ref-core/src/climate_ref_core/esgf/cmip7.py [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+def _get_cmip7_cache_dir() -> Path:
+    """Get the cache directory for converted CMIP7 files."""
+    # Use platform-appropriate cache directory for climate-ref
+    # This avoids polluting the intake-esgf cache with converted files
+    cache_dir = Path(platformdirs.user_cache_dir("climate-ref")) / "cmip7-converted"
+    cache_dir.mkdir(parents=True, exist_ok=True)
+    return cache_dir
+
+
+def _convert_file_to_cmip7(cmip6_path: Path, cmip7_facets: dict[str, str]) -> Path:
```
[the remark this answers is not in the document]
Comment (lewisjared) [reply packages/climate-ref-core/src/climate_ref_core/esgf/cmip7.py]: Fixed in 97242bf6 - Updated type annotation to dict[str, Any].
